File: dart/2015/day02a/day02a.dart
#!/usr/bin/env dart

import 'dart:io';
import 'dart:math';

void usage(String progName) {
    print('usage: $progName <input file>');
    exit(1);
}

int process(String filename) {
    int totalArea = 0;

    final lines = File(filename).readAsLinesSync();
    for ( final line in lines ) {
        List<String> strDims = line.split('x');
        int l = int.parse(strDims[0]);
        int w = int.parse(strDims[1]);
        int h = int.parse(strDims[2]);
        int area1 = l * w;
        int area2 = l * h;
        int area3 = w * h;
        int surfaceArea = 2 * area1 + 2 * area2 + 2 * area3;
        int minArea = min(area1, min(area2, area3));
        totalArea += surfaceArea + minArea;
    }
    return totalArea;
}

void main(List<String> args) {
    if ( args.isEmpty ) {
        usage(Platform.script.pathSegments.last);
    }
    final filename = args[0];
    final result = process(filename);
    print('result = $result');
}

File: julia/2015/day02a/day02a.jl
#!/usr/bin/env julia

function usage()
    progname = PROGRAM_FILE
    println(stderr, "usage: $progname <input file>")
    exit(1)
end

function process(content::String)::Int32
    total_area = 0
    for line in split(content, '\n', keepempty=false)
        dims = parse.(Int32, split(line, 'x'))
        l, w, h = dims
        area1 = l * w
        area2 = l * h
        area3 = w * h
        surface_area = 2 * area1 + 2 * area2 + 2 * area3
        min_area = minimum([area1, area2, area3])
        total_area += surface_area + min_area
    end
    return Int32(total_area)
end

function main()
    args = ARGS
    if length(args) < 1
        usage()
    end
    filename = args[1]
    try
        content = read(filename, String)
        result = process(content)
        println("result = $result")
    catch e
        println(stderr, "error reading file: $e")
        exit(1)
    end
end

if abspath(PROGRAM_FILE) == @__FILE__
    main()
end
